### ISSUE TITLE
[Enhancement] Run scan based on Jar list

### ISSUE THREAD

**Issue by rocketbootkid** [dependency-check/DependencyCheck#627]
Currently, you need to point DC at a folder full of the files you want to scan. What would be easier would be to simply provide a list of the files you are using, rather than go through the rigmarole of asking complicated questions of your build envt to get all the files in once place.

If the filename is all that's required to search the NVD, why do we need the whole file? I guess it's perhaps because you CRC the file and use that to lookup the NVD? 

If not, can we have an option to scan using a list instead?

**Comment by jeremylong** [comment]
Dependency-check does more then obtain the hash from the actual JAR file; consider reading [How dependency-check works](http://jeremylong.github.io/DependencyCheck/general/internals.html). Just using a list of dependency names would not provide enough information for accurate identification in many cases.

Have you considered instead of "going through the rigmarole of asking complicated questions of you build environment" just using one of the dependency-check build plugins? There are Maven, gradle, Ant, and sbt plugins.

**Comment by rocketbootkid** [comment]
I had looked for "how it works" info, but clearly not well enough :-)

I'm currently automating AppScan Source scans with Jenkins, and was wanting to bolt on a DC scan as well. Since AppScan requires dependencies, I am using Maven to generate them, so that's not an issue. I call the command line DC and point at the folder of dependencies I already have.

We are using the DC Maven plugin as part of our CI, but it can add time to builds so I was looking at decoupling it.

In short, I was trying to make it as simple and lightweight as possible. Our auto deploy system generates a list of Jars, so I was hoping to be able to use that rather than have to actually compile a folder of Jars.

Thanks.

**Comment by jeremylong** [comment]
The maven plugin build time generally comes from having to update the local DB (a copy of the NVD CVE data). One thing some orgs have done is setup a [central database](http://jeremylong.github.io/DependencyCheck/data/database.html), setup a single Jenkins job to update the database, and then run all other DC scans by pointing to the central database with the noupdate flag set.

As I believe your question has been answered I'm going to close this ticket. Feel free to re-open it if I am mistaken.
